lib/src/public_interface.dart

```diff
@@ -0,0 +1,166 @@
+  /// doc.parseValueAt(['b', 'd']); // ArgumentError
+  /// doc.parseValueAt(['b']); // null
+  /// doc.parseValueAt(['c']); // YamlScalar(null)
+  ///
+  /// final doc2 = YamlEditor('[0,1]');
+  /// doc2.parseValueAt([2]); // ArgumentError
+  /// doc2.parseValueAt(["2"]); // ArgumentError
+  /// ```
+  YamlNode parseValueAt(Iterable<Object> path, {Object Function() orElse});
+
+  /// Sets [value] in the [path]. Takes an optional [style] parameter.
+  ///
+  /// If the [path] is not accessible (e.g. it currently does not exist in the document),
+  /// an error will be thrown. Note that [setIn] provides a different result as compared to
+  /// a [removeIn] followed by an [insertIn], because it preserves comments at the same level.
+  ///
+  /// ```dart
+  /// final doc = YamlEditor('''
+  ///   - 0
+  ///   - 1 # comment
+  ///   - 2
+  /// ''');
+  /// doc.setIn([1], 'test');
+  /// ```
+  ///
+  /// Expected Output:
+  /// '''
+  ///   - 0
+  ///   - test # comment
+  ///   - 2
+  /// '''
+  ///
+  /// ```dart
+  /// final doc2 = YamlEditor("[YAML Ain't Markup Language   # comment]");
+  /// doc2.removeIn([1]);
+  /// doc2.insertInList([1], 'test');
+  /// ```
+  ///
+  /// Expected Output:
+  /// '''
+  ///   - 0
+  ///   - test
+  ///   - 2
+  /// '''
+  void setIn(Iterable<Object> path, {YamlStyle style});
+
+  /// Removes the value in the path.
+  void removeIn(Iterable<Object> path);
+
+  /// Appends [value] into the list at [listPath], only if the element at the given path
+  /// is a List.
+  ///
+  /// Users have the option of defining the indentation applied and whether
+  /// flow structures will be applied via the optional parameter [style]. For a comprehensive
+  /// list of styling options, refer to the documentation for [YamlStyle].
+  ///
+  /// **Convenience Method**
+  /// [addInList] is equivalent to [insertInList] with index = length.
+  void addInList(Iterable<Object> listPath, Object value,
+      {YamlStyle yamlStyle});
+
+  /// Prepends [value] into the list at [listPath], only if the element at the given path
+  /// is a List.
+  ///
+  /// Users have the option of defining the indentation applied and whether
+  /// flow structures will be applied via the optional parameter [style]. For a comprehensive
+  /// list of styling options, refer to the documentation for [YamlStyle].
+  ///
+  /// **Convenience Method**
+  /// [prependInList] is equivalent to [insertInList] with index = 0.
+  void prependInList(Iterable<Object> listPath, Object value,
+      {YamlStyle yamlStyle});
+
+  /// Inserts [value] into the list at [listPath], only if the element at the given path
+  /// is a list. [index] must be non-negative and no greater than the list's length.
+  ///
+  /// Users have the option of defining the indentation applied and whether
+  /// flow structures will be applied via the optional parameter [style]. For a comprehensive
+  /// list of styling options, refer to the documentation for [YamlStyle].
```

Comment:
Does this overwrite the existing element in the list at said location?

Reply:
nope! 

```dart
      final doc = YamlEditor('''
- 0 # comment a
- 2 # comment b
''');
      doc.insertInList([], 1, 1);
      expect(doc.toString(), equals('''
- 0 # comment a
- 1
- 2 # comment b
'''));
```